common/models/templates/src/trie.ts

```diff
@@ -88,12 +98,26 @@ export class TrieTraversal implements LexiconTraversal {
     return traversal;
   }
 
+  private sortNodeIfNeeded(node: Node) {
```

Comment:
I found this by referencing the _original_ original implementation:

https://github.com/keymanapp/keyman/blob/dd12e7315353d7d8276825f36be6fe29cb796abd/developer/src/kmc-model/src/build-trie.ts#L207-L208

https://github.com/keymanapp/keyman/blob/dd12e7315353d7d8276825f36be6fe29cb796abd/developer/src/kmc-model/src/build-trie.ts#L234

This may seem "pulled up a _tad_ bit too early", but there's a reason.  If we 'construct' the Trie while live, modifications can make part of the Trie unsorted.  We want to track that and correct for that.

The followup then changes _this_ block, integrating it within the Trie traversal structure.  Pulling it in early like this allows for code comparison between the two approaches.